download_collection.py

```diff
@@ -0,0 +1,722 @@
+        scaleCoeff = 65535.0 / zFar
+
+        depth_buf = np.ones((height, width), dtype=np.float32) * zFar
+        color_buf = np.zeros((height, width, 3), dtype=np.float32)
+
+        settings = cv.TriangleRasterizeSettings().setShadingType(cv.RASTERIZE_SHADING_SHADED)
+        settings = settings.setCullingMode(cv.RASTERIZE_CULLING_NONE)
+
+        color_buf, depth_buf = cv.triangleRasterize(verts, indices, colors, color_buf, depth_buf,
+                                                    cameraPose, fovY, zNear, zFar, settings)
+
+        if verbose:
+            print("rasterized")
+
+        if doRemap:
+            remap_color_buf, _ = cv.triangleRasterizeColor(verts, indices, remapCoords, color_buf, depth_buf,
+                                                           cameraPose, fovY, zNear, zFar, settings)
+            mapx = remap_color_buf[:, :, 0] * texture.shape[1] - 0.5
+            mapy = remap_color_buf[:, :, 1] * texture.shape[0] - 0.5
+            remapped = cv.remap(texture, mapx, mapy, cv.INTER_LINEAR)
+            cv.imwrite(remap_path, remapped * 255.0)
+
+        colorRasterize = color_buf
+        depthRasterize = (depth_buf * scaleCoeff)
+
+        if debug_pics_dir:
+            cv.imwrite(debug_color_path, color_buf * 255.0)
+            cv.imwrite(debug_depth_path, depthRasterize.astype(np.ushort))
+
+        cv.imwrite(color_raster_path, color_buf * 255.0)
+        cv.imwrite(depth_raster_path, depthRasterize.astype(np.ushort))
+
+        # send mesh to OpenGL rasterizer
+
+        renderer_args = [renderer_path] + [
+                "--modelPath="+str(colvert_path),
+                "--custom",
+                "--fov="+str(fovDegrees),
+                "--posx="+str(position[0]),
+                "--posy="+str(position[1]),
+                "--posz="+str(position[2]),
+                "--lookatx="+str(lookat[0]),
+                "--lookaty="+str(lookat[1]),
+                "--lookatz="+str(lookat[2]),
+                "--upx="+str(upVector[0]),
+                "--upy="+str(upVector[1]),
+                "--upz="+str(upVector[2]),
+                "--resx="+str(width),
+                "--resy="+str(height),
+                "--zNear="+str(zNear),
+                "--zFar="+str(zFar),
+                "--scaleCoeff="+str(scaleCoeff),
+                # white/flat/shaded
+                "--shading=shaded",
+                # none/cw/ccw
+                "--culling=cw",
+                "--colorPath="+str(rgb_gl_path),
+                "--depthPath="+str(depth_gl_path),
+            ]
+
+        if verbose:
+            print(renderer_args)
+        p = subprocess.run(renderer_args)
+
+        # compare results
+
+        colorGl = cv.imread(rgb_gl_path)
+        colorGl = colorGl.astype(np.float32) * (1.0/255.0)
+        colorDiff = np.ravel(colorGl - colorRasterize)
+        normInfRgb = np.linalg.norm(colorDiff, ord=np.inf)
+        normL2Rgb = np.linalg.norm(colorDiff, ord=2) / (width * height)
```

Comment:
I propose to report min/max/average, but not only value normalized on pixel count. In case if the scene is sparse and most of pixels are background the current value is not informative.